src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionDuplicatedArgsInspection.kt

```diff
@@ -30,18 +31,18 @@ class SmkSectionDuplicatedArgsInspection : SnakemakeInspection() {
         ) {
             val args = argumentList?.arguments ?: emptyArray()
             if (args.size > 1) {
-                val setOfDeclaredArguments = mutableSetOf<String>()
+                val setOfDeclaredArguments = mutableSetOf<PsiElement>()
 
                 args.forEach { arg ->
-                    if (arg.text in setOfDeclaredArguments) {
+                    if (setOfDeclaredArguments.any{x -> arg.textMatches(x)}) {
```

Comment:
I've just checked `arg.textMatches` sources and was a bit surprised. Maybe it is ok for a single usage or when psi element is PsiFile/PsiDirectory, but not for our case. Your prev impl using set was better. Please read `textMatches` implementation in IDEA sources + you could check actual behavior in  a debugger.

Reply:
Done in d3530b55a64b2c25d87b8154f14ca225a634a17a